docs/rules/object-key-newlines.md

```diff
@@ -0,0 +1,68 @@
+# Enforce placing object keys on separate lines (object-key-newlines)
+
+While formatting preferences are very personal, a number of style guides require that
```

Comment:
> @vitorbal Based on #5933 (comment) and your confirmation about the example, renaming the rule as object-property-newline seems solid.

Alright, renamed it to `object-property-newline`. Renamed the option name as well to `allowMultiplePropertiesPerLine`.

> 
> Super point about diffs! See how the introduction reads with just a sentence about that and adapt those introduction examples to be parallel with the example code. Just confirmed that diff formatting works in Jekyll and Rouge.

Done, let me know what you think of the new suggestion!